waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -220,3 +225,26 @@ genRoutesDir spec =
 
 operationsRouteInRootRouter :: String
 operationsRouteInRootRouter = "operations"
+
+depsRequiredByPassport :: AppSpec -> [App.Dependency.Dependency]
+depsRequiredByPassport spec =
+  AS.Dependency.fromList $
+    concat
+      [ [("passport", "0.6.0") | externalAuthEnabled],
+        [("passport-google-oauth20", "2.0.0") | googleAuthEnabled]
+      ]
+  where
+    maybeAuth = AS.App.auth $ snd $ getApp spec
+    externalAuthEnabled = maybe False AS.App.Auth.isExternalAuthEnabled maybeAuth
+    googleAuthEnabled = maybe False AS.App.Auth.isGoogleAuthEnabled maybeAuth
+
+genPatches :: AppSpec -> Generator [FileDraft]
+genPatches spec = patchesRequiredByPassport spec
+
+patchesRequiredByPassport :: AppSpec -> Generator [FileDraft]
+patchesRequiredByPassport spec =
+  return $
+    [C.mkTmplFd (C.asTmplFile [relfile|patches/oauth+0.9.15.patch|]) | externalAuthEnabled]
+  where
+    maybeAuth = AS.App.auth $ snd $ getApp spec
+    externalAuthEnabled = maybe False AS.App.Auth.isExternalAuthEnabled maybeAuth
```

Comment:
Sounds like we could externalize this one as a function in ServerGenerator? Or maybe in ServerGenerator.Common?

Reply:
Yeah, we could pull this out. It is used in both Server and Web App contexts though, but doesn't feel important enough to be Generator.Common to me... maybe I'll just add a helper to ServerGenerator for now so we can reduce the duplication in the file?

Reply:
Ok, whatever you think is the best! I thought it could be at level of Generator, because it is not specific for ServerGenerator.

Reply:
Ha, they do feel a bit repetitive and redundant! But they do no harm. I get what you mean to do. I guess alternative could be to define a function `fromMaybeBool :: Maybe Bool -> Bool` and then use that where needed. You could also just replace `isGoogleAuthEnabled auth` with `(isGoogleAuthEnabled <$> auth) == Just True` where needed, to obtain the same effect.